lightning-background-processor/src/lib.rs

```diff
@@ -129,10 +138,12 @@ impl BackgroundProcessor {
 			let mut current_time = Instant::now();
 			loop {
 				peer_manager.process_events();
+				channel_manager.process_pending_events(&event_handler).unwrap(); // TODO: Return the error
+				chain_monitor.process_pending_events(&event_handler).unwrap(); // TODO: Return the error
```

Comment:
There's an open question around how to handle errors in `BackgroundProcessor`. Currently, it will simply `unwrap` and fail if there is an error. Alternatively, it could return the error in the same manner as it does for the persister, which would necessitate defining an error type covering both.